src/Microsoft.ComponentDetection.Detectors/pip/ISimplePyPiClient.cs

```diff
@@ -0,0 +1,310 @@
+{
+    // Values used for cache creation
+    private const long CACHEINTERVALSECONDS = 60;
+    private const long DEFAULTCACHEENTRIES = 4096;
+
+    // max number of retries allowed, to cap the total delay period
+    public const long MAXRETRIES = 15;
+
+    private static readonly HttpClientHandler HttpClientHandler = new HttpClientHandler() { CheckCertificateRevocationList = true };
+    private static readonly ProductInfoHeaderValue ProductValue = new(
+    "ComponentDetection",
+    Assembly.GetEntryAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion);
+
+    private static readonly ProductInfoHeaderValue CommentValue = new("(+https://github.com/microsoft/component-detection)");
+
+    // time to wait before retrying a failed call to pypi.org
+    private static readonly TimeSpan RETRYDELAY = TimeSpan.FromSeconds(1);
+
+    private readonly IEnvironmentVariableService environmentVariableService;
+    private readonly ILogger<SimplePyPiClient> logger;
+
+    // Keep telemetry on how the cache is being used for future refinements
+    private readonly SimplePypiCacheTelemetryRecord cacheTelemetry;
+
+    /// <summary>
+    /// A thread safe cache implementation which contains a mapping of URI -> SimpleProject for simplepypi api projects
+    /// and has a limited number of entries which will expire after the cache fills or a specified interval.
+    /// </summary>
+    private MemoryCache cachedSimplePyPiProjects = new MemoryCache(new MemoryCacheOptions { SizeLimit = DEFAULTCACHEENTRIES });
+
+    /// <summary>
+    /// A thread safe cache implementation which contains a mapping of URI -> Stream for project wheel files
+    /// and has a limited number of entries which will expire after the cache fills or a specified interval.
+    /// </summary>
+    private MemoryCache cachedProjectWheelFiles = new MemoryCache(new MemoryCacheOptions { SizeLimit = DEFAULTCACHEENTRIES });
+
+    private bool checkedMaxEntriesVariable;
+
+    // retries used so far for calls to pypi.org
+    private long retries;
+
+    public SimplePyPiClient() => this.cacheTelemetry = new SimplePypiCacheTelemetryRecord()
+    {
+        NumProjectFileCacheHits = 0,
+        NumSimpleProjectCacheHits = 0,
+        FinalProjectFileCacheSize = 0,
+        FinalSimpleProjectCacheSize = 0,
+    };
+
+    public SimplePyPiClient(IEnvironmentVariableService environmentVariableService, ILogger<SimplePyPiClient> logger)
+    {
+        this.environmentVariableService = environmentVariableService;
+        this.cacheTelemetry = new SimplePypiCacheTelemetryRecord
+        {
+            NumProjectFileCacheHits = 0,
+            NumSimpleProjectCacheHits = 0,
+            FinalProjectFileCacheSize = 0,
+            FinalSimpleProjectCacheSize = 0,
+        };
+        this.logger = logger;
+    }
+
+    public static HttpClient HttpClient { get; internal set; } = new HttpClient(HttpClientHandler);
```

Comment:
Is it possible to inject this with dependency injection? See https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests